lib/Promise/ES6.pm

```diff
@@ -415,7 +415,7 @@ sub _is_promise {
 }
 
 sub DESTROY {
-    return if $$ != $_[0]{'_pid'};
+    return unless $_[0]{'_pid'} && $$ == $_[0]{'_pid'};
```

Comment:
Ordinarily I’m averse to `unless`, but here, at least, I can agree it seems at least as clear. <shrug>